src/compiler/binder.ts

```diff
@@ -2521,7 +2521,14 @@ namespace ts {
                         // Declare a 'member' if the container is an ES5 class or ES6 constructor
                         constructorSymbol.members = constructorSymbol.members || createSymbolTable();
                         // It's acceptable for multiple 'this' assignments of the same identifier to occur
-                        declareSymbol(constructorSymbol.members, constructorSymbol, node, SymbolFlags.Property, SymbolFlags.PropertyExcludes & ~SymbolFlags.Property);
+                        if (hasDynamicName(node)) {

```

Comment:
this is such a complicated, yet nearly identical, addition in all 3 places, I wonder if it's possible to de-dupe the code at all. At least the first two changes are basically the same.

Reply:
Yeah, I was thinking the same thing when I started to write the 3rd one.